document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/handler/CompanyReportDataHandler.java

```diff
@@ -0,0 +1,134 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static uk.gov.companieshouse.document.generator.company.report.CompanyReportDocumentInfoServiceImpl.MODULE_NAME_SPACE;
+
+@Component
+public class CompanyReportDataHandler {
+
+    @Autowired
+    private CompanyService companyService;
+
+    @Autowired
+    private CompanyReportMapper companyReportMapper;
+
+    private static final Logger LOG = LoggerFactory.getLogger(MODULE_NAME_SPACE);
+
+    public DocumentInfoResponse getCompanyReport(String resourceUri, String requestId)
+        throws HandlerException {
+
+        String companyNumber = getCompanyNumberFromUri(resourceUri);
+
+        try {
+            LOG.infoContext(requestId, "Getting data for report for company number: " + companyNumber, getDebugMap(companyNumber));
+            return createDocumentInfoResponse(companyNumber);
+        } catch (URIValidationException | ApiErrorResponseException | MapperException e) {
+            LOG.errorContext(requestId,"Failed to get data for report for company number " + companyNumber, e, getDebugMap(companyNumber));
+            throw new HandlerException(e.getMessage(), e.getCause());
+        }
+    }
+
+    private DocumentInfoResponse createDocumentInfoResponse(String companyNumber) throws HandlerException,
+        URIValidationException, ApiErrorResponseException, MapperException {
+
+        DocumentInfoResponse documentInfoResponse = new DocumentInfoResponse();
+
+        documentInfoResponse.setData(getCompanyReportData(companyNumber));
+        documentInfoResponse.setAssetId("company-report");
+        documentInfoResponse.setPath(createPathString());
+        documentInfoResponse.setTemplateName("company-report.html");
+
+        return documentInfoResponse;
+    }
+
+    private String getCompanyReportData(String companyNumber) throws HandlerException,
+        URIValidationException, ApiErrorResponseException, MapperException {
+
+        CompanyReportApiData companyReportApiData = new CompanyReportApiData();
+
+        CompanyProfileApi companyProfileApi = getCompanyProfile(companyNumber);
+
+        companyReportApiData.setCompanyProfileApi(companyProfileApi);
+
+        return toJson(companyReportMapper.mapCompanyReport(companyReportApiData));
+    }
+
+
+    private String toJson(CompanyReport companyReport) throws HandlerException {
+
+        String reportToJson;
+        ObjectMapper mapper = new ObjectMapper();
+        JavaTimeModule javaTimeModule = new JavaTimeModule();
+        javaTimeModule.addDeserializer(LocalDate.class, new LocalDateDeserializer(DateTimeFormatter.ofPattern("dd-MMMM-yyyy")));
+        javaTimeModule.addSerializer(LocalDate.class, new LocalDateSerializer(DateTimeFormatter.ofPattern("dd-MMMM-yyyy")));
+        mapper.registerModule(javaTimeModule);
+        mapper.setDateFormat(new SimpleDateFormat("dd-MMMM-yyyy"));
+
+        try {
+            LOG.info("Attempting to convert company report to JSON");
```

Comment:
swap to infoContext and add requestId and a debug map so Logs can be tracked for the specific request